Initial support ip-address-type as annotation

njuettner opened the pull request:
Support using the annotation "alb.ingress.kubernetes.io/ip-address-type" to change the ALB for dualstack (ipv4 by default)

njuettner commented:
Enabling the VPC to support IPv6 and attaching to each subnet IPv6 CIDR blocks, it works really well:

Got created/updated/deleted a dualstack ALB successfully.

```
host kube-ing-LB-1AZ0O87ULFKQI-753189733.eu-central-1.elb.amazonaws.com
kube-ing-LB-1AZ0O87ULFKQI-753189733.eu-central-1.elb.amazonaws.com has address 35.156.227.194
kube-ing-LB-1AZ0O87ULFKQI-753189733.eu-central-1.elb.amazonaws.com has IPv6 address 2a05:d014:9c0:bf67:2717:e4e0:2e90:3894
```

szuecs commented on kubernetes/adapter.go:
```diff
@@ -99,6 +100,11 @@ func (a *Adapter) newIngressFromKube(kubeIngress *ingress) *Ingress {
 	}
 
+	ipAddressType := "ipv4"
+	if kubeIngress.getAnnotationsString(ingressALBIPAdressType, "") == "dualstack" {
```
`"dualstack"` should be a const

szuecs commented on kubernetes/ingress.go:
```diff
@@ -59,6 +59,7 @@ type ingressLoadBalancer struct {
 }
 
 const (
+	ingressALBIPAdressType          = "alb.ingress.kubernetes.io/ip-address-type"
```
Please change to `zalando.org/ip-address-type`, because annotations are specific to the tool.

szuecs commented:
@njuettner Thanks for the PR.
This change also requires to be added to the documentation.
Please update the readme.

njuettner commented:
@szuecs addressed you comments, thanks for the review! I've moved example to docs and also added a annotation file in there. I added other annotations as well but I don't know if there are more to add there. This is the same approach how **aws-alb-ingress-controller** documented it.

szuecs commented:
@njuettner please do not break paths in this PR `example/` to `docs/example/` change is not part of this PR and needs a decision, if we want to do this. There are already quite many links spread in a bunch of projects that link to these.

szuecs commented on docs/annotations.md:
```diff
@@ -0,0 +1,14 @@
+# Ingress annotations
```
please put this into the README.md

szuecs commented:
:+1:

mikkeloscar commented:
:+1:
